app/controllers/tasks_controller.rb

```diff
@@ -122,13 +122,15 @@ def move_up
     end
   end
   def show
+    task_id=params[:id].to_i
     @users=User.all
     @accessible_tasks=@user.shared_tasks.all(:select => "DISTINCT(task_shares.task_id),task_shares.position,tasks.*",:joins => 'INNER  JOIN task_shares ts  ON task_shares.task_id = tasks.id',:order => "task_shares.position")
     # @task = @accessible_tasks[params[:id].to_i]
-    @task=@accessible_tasks.detect{|x| x.task_id==params[:id]}
+    @task=@accessible_tasks.detect{|x| x.task_id==task_id}
```

Comment:
Surrounding space missing for operator '='.<br>Surrounding space missing for operator '=='.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.